### DISCUSSION TITLE
WSL Register Distribution Failed

### DISCUSSION THREAD

**Discussion by Squidility** [Cyfrin/foundry-full-course-cu#80] [Q&A] [answered]
Hi 
I have been trying to install ubuntu on windows wsl using the method on the youtube video but I keep getting the following error:
```
Installing, this may take a few minutes...
WslRegisterDistribution failed with error: 0x80370102
Please enable the Virtual Machine Platform Windows feature and ensure virtualization is enabled in the BIOS.
For information please visit https://aka.ms/enablevirtualization
Press any key to continue...
```

After two hours of google searching and chatting to gpt I have found that Windows 10 Home may not be capable of running the WSL. Firstly, may I ask if anyone knows a way to make it work; or, is there another open source product I could use?

**Comment by devblixt** [comment]
Hey, 
Windows 10 is capable of running WSL. You have to enable the Virtual Machine Platform Windows feature. For this, go to Control Panel (Not system settings) and go to the programs and features section. You'll find a button to 'Turn Windows features on or off'. Click on the button and you can find options to enable Virtual Machine Platform.

**Comment by Squidility** [comment, marked as answer]
After hours of searching it ended up being disabled in my bios. The EZ section of the MSI bios indicated that virtualization was enabled. However, I found that in advanced mode it was clearly disabled. For anyone else having this issue, please quadruple check your bios settings.
